core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -135,49 +130,53 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
             } yield {
-              RequestComplete(StatusCodes.OK, WorkspaceResponse(accessLevel, canShare, canCompute, catalog, workspaceContext.workspace, stats, owners))
+              RequestComplete(StatusCodes.OK, WorkspaceResponse(accessLevel, canShare, canCompute, canCatalog, workspaceContext.workspace, stats, owners))
             }
           }
         }
       }
     }
 
-  def getMaximumAccessLevel(user: RawlsUser, workspaceContext: SlickWorkspaceContext, dataAccess: DataAccess): ReadWriteAction[WorkspaceAccessLevel] = {
-    val accessLevels = workspaceContext.workspace.authDomainACLs.map { case (accessLevel, groupRef) =>
-      dataAccess.rawlsGroupQuery.loadGroupIfMember(groupRef, user).map {
-        case Some(_) => accessLevel
-        case None => WorkspaceAccessLevels.NoAccess
-      }
-    }
+  def getUserComputePermissions(workspaceId: String, userAccessLevel: WorkspaceAccessLevel): Future[Boolean] = {
+    if(userAccessLevel >= WorkspaceAccessLevels.Owner) Future.successful(true)
+    else samDAO.userHasAction(SamResourceTypeNames.workspace, workspaceId, SamResourceActions.launchBatchCompute, userInfo)
+  }
 
-    DBIO.sequence(accessLevels).map { _.fold(WorkspaceAccessLevels.NoAccess)(WorkspaceAccessLevels.max) }
+  def getUserSharePermissions(workspaceId: String, userAccessLevel: WorkspaceAccessLevel, accessLevelToShareWith: WorkspaceAccessLevel): Future[Boolean] = {
+    if(userAccessLevel >= WorkspaceAccessLevels.Owner) Future.successful(true)
+    else samDAO.userHasAction(SamResourceTypeNames.workspace, workspaceId, SamResourceActions.sharePolicy(accessLevelToShareWith.toString.toLowerCase), userInfo)
   }
 
-  def getWorkspaceOwners(workspace: Workspace, dataAccess: DataAccess): ReadWriteAction[Seq[String]] = {
-    dataAccess.rawlsGroupQuery.load(workspace.accessLevels(WorkspaceAccessLevels.Owner)).flatMap {
-      case None => DBIO.failed(new RawlsException(s"Unable to load owners for workspace ${workspace.toWorkspaceName}"))
-      case Some(ownerGroup) =>
-        val usersAction = DBIO.sequence(ownerGroup.users.map(dataAccess.rawlsUserQuery.load(_).map(_.get.userEmail.value)).toSeq)
-        val subGroupsAction = DBIO.sequence(ownerGroup.subGroups.map(dataAccess.rawlsGroupQuery.load(_).map(_.get.groupEmail.value)).toSeq)
+  def getUserCatalogPermissions(workspaceId: String): Future[Boolean] = {
+    samDAO.userHasAction(SamResourceTypeNames.workspace, workspaceId, SamResourceActions.workspaceCanCatalog, userInfo)
+  }
 
-        for {
-          users <- usersAction
-          subGroups <- subGroupsAction
-        } yield users ++ subGroups
+  def getMaximumAccessLevel(workspaceId: String): Future[WorkspaceAccessLevel] = {
+    for {
+      isOwner <- samDAO.userHasAction(SamResourceTypeNames.workspace, workspaceId, SamResourceActions.workspaceOwn, userInfo).map(hasAction => if(hasAction) Some(WorkspaceAccessLevels.Owner) else None)
```

Comment:
there's probably a more elegant way to write this but the _most_ clean way to write it would be to have an endpoint that returns all of a user's actions for a resource. then we only need one call to sam.